utils/tasks.js

```diff
@@ -0,0 +1,54 @@
+
+const { getUsername, getUserId, getParticipantUsernames, getParticipantUserIds } = require('./users')
+
+const fromFirestoreData = async (task) => {
+  if (!task) {
+    return task
+  }
+  let { ownerId, createdBy, assignedTo, participants } = task
+  if (ownerId) {
+    ownerId = await getUsername(ownerId)
+  }
+  if (createdBy) {
+    createdBy = await getUsername(createdBy)
+  }
+  if (assignedTo) {
+    assignedTo = await getUsername(assignedTo)
+  }
+
+  if (participants) {
+    participants = await getParticipantUsernames(participants)
+  }
+
+  return {
+    ...task,
+    ownerId,
+    createdBy,
+    assignedTo,
+    participants
+  }
+}
+
+const toFirestoreData = async (task) => {
+  if (!task) {
+    return task
+  }
+  const updatedTask = { ...task }
+  const { ownerId, assignedTo, participants } = task
+  if (ownerId) {
+    updatedTask.ownerId = await getUserId(ownerId)
+  }
+  if (assignedTo) {
+    updatedTask.assignedTo = await getUserId(assignedTo)
+  }
+
+  if (participants) {
+    updatedTask.participants = await getParticipantUserIds(participants)
+  }
+  return updatedTask
+}
+
+module.exports = {
+  fromFirestoreData,
+  toFirestoreData
```

Comment:
These function names don't make much sense 🤔 
Can you please change them to something like `getUserIdsForTask` and `getUsernamesForTask`

Reply:
I guess it depends on developer's background. :). It is a very common practice to translate database objects to domain objects and vice versa. This usually happens in 2 common functions (toDao and fromDao).
The idea is all field translations in one place rather than creating mulitple functions for each fields.

E.g.
Even though the current functions only care about userId and usernames, suppose you have a new field with a date type, but you want to store them in DB as a timestamp, you only need to add those translations in these functions and not worry about any other changes CRUD operation changes. 

Does this help?